### PR TITLE
Create tsp, migrate test-proxy swagger to tsp

### PR DESCRIPTION
**Pull request by scbedd**
Addresses #9450

### CODE REVIEW

**Review by mikeharder — APPROVED** [review]
I never even knew test-proxy had a swagger spec!

**Review by benbp — APPROVED** [review]

**Comment by benbp** [comment]
> I never even knew test-proxy had a swagger spec!

@mikeharder me neither!

File: tools/test-proxy/typespec/main.tsp [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    op start(@body body: StartPayload): {
+        @statusCode statusCode: 200;
+        @header("x-recording-id") recordingId: string;
+    };
+}
+
+@route("/Record/Stop")
+namespace RecordStop {
+    model VariablePayload {
+        Variables: Record<string>;
```
Comment (scbedd) [inline tools/test-proxy/typespec/main.tsp:31]: I need this to be `Dictionary<string, string>`. I know typescript supports `Record<T, V>`, so I'm a bit weirded out by this. It's a typespec specific issue.

**Comment by heaths** [comment]
@lmazuel since this is in a different repo, what do I need to do to generate a client on it? Just do it manually? It changes little enough I'm fine with that, but want to check it's the right thing to do. Or can tsp-client.yml handle other repos as well?

**Pull request merged**